frontend/components/product-list/sections/FilterableProductsSection/RangeInput.tsx

```diff
@@ -0,0 +1,219 @@
+import {
+   Box,
+   FormControl,
+   FormErrorMessage,
+   FormLabel,
+   HStack,
+   Input,
+   InputGroup,
+   InputLeftElement,
+   Text,
+} from '@chakra-ui/react';
+import { formatFacetName } from '@helpers/algolia-helpers';
+import { useDebounce } from '@ifixit/ui';
+import * as React from 'react';
+import { useRange, UseRangeProps } from 'react-instantsearch-hooks-web';
+import { useSearchCache } from './useSearchCache';
+
+export type RangeInputProps = UseRangeProps;
+
+const DEBOUNCE_INTERVAL_MILLIS = 300;
+
+export function RangeInput(props: RangeInputProps) {
+   const { refresh } = useSearchCache();
+   // This is a quick hack to get `useRange` to work. Without this, `useRange`
+   // would behave like price is not refinable. Eventually we should figure out
+   // how to make `useRange` work without this hack.
+   React.useEffect(() => {
+      refresh();
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, []);
+
+   const range = useRange(props);
+   const appliedRangeMin = range.start[0];
+   const appliedRangeMax = range.start[1];
+
+   const refineMinValue = React.useCallback(
```

Comment:
`refineMinValue` and `refineMaxValue` don't appear consistent with each other. I'm confused about a couple things:

- Why do we use `clamp(value, 0, maxAllowedValue)` here, but in refineMaxValue we do `clamp(value, undefined, maxAllowedValue)`? I would expect `clamp(value, minAllowedValue, Infinity)` there.
- Here, we use `clampedValue < appliedRangeMax` in isValidRange, but in refineMaxValue we use `clampedValue > minAllowedValue && clampedValue > appliedRangeMin` in isValidRange. I would expect only `clampedValue > minAllowedValue` there.
- the call to `range.refine(...)` is different here than in refineMaxValue

If any of these require changes, I'm also happy to have it happen in a followup pull.

Reply:
I think our code should not try to be "smart" about it. Let the user type whatever they want, no auto-fixing. If they type something nonsensical like "10 to 5" (instead of `5 to 10`) we can just ignore the price filter or show no results, both seem like reasonable behavior.